logger/logger.go

```diff
@@ -64,6 +64,10 @@ type logger struct {
 	Logger kitlog.Logger
 }
 
+func (l *logger) Decorate(keyvals ...interface{}) {
+	l.Logger = kitlog.NewContext(l.Logger).With(keyvals...)
```

Comment:
We are using an older version of `kit-go`, `With` is [defined on Context](https://github.com/go-kit/kit/blob/f66b0e13579bfc5a48b9e2a94b1209c107ea1f41/log/log.go#L90).

As to the copying approach, I don't know if that would solve the problems @xh3b4sd is alluding to (although it's not yet clear to me what those problems are).